lnst/Recipes/ENRT/ConfigMixins/LongLivedConnectionsMixin.py

```diff
@@ -0,0 +1,230 @@
+import time
+import logging
+import signal
+import copy
+from math import ceil
+from lnst.Tests.LongLivedConnections import LongLivedServer, LongLivedClient
+from lnst.Common.IpAddress import Ip6Address, Ip4Address
+from lnst.Common.IpAddress import interface_addresses
+from lnst.Common.Parameters import IntParam
+from lnst.Common.IpAddress import Ip4Address, Ip6Address, BaseIpAddress
+from lnst.Recipes.ENRT.helpers import ip_endpoint_pairs
+from lnst.Common.IpAddress import ip_version_string
+
+
+class LongLivedConnectionsMixin:
+    """
+    This mixin adds support for long-lived connections.
+
+    Based on `long_lived_conns` parameter, it will create a number of
+    long-lived connections between the hosts. Receiver is a server, while
+    generator is a client.
+
+    Only long lived connections IPs are handled by this mixin.
+    Therefore, if your test requires perf IPs, it should be
+    configured by parent's test_wide_configuration() method.
+    IPs used for long-lived client and server are added based on
+    long_lived_conns_per_ip parameter, which defines size of
+    addressable space for connections. Sinde L4 can address up
+    to 65535 ports, it's obviously limited by that.
+
+    Connections are not equally distributed among clients.
+    The first client (and others) will get long_lived_conns_per_ip
+    connections, while the last one will get the remaining connections.
+
+    Don't forget to set appropriate NO_FILES ulimit (if needed).
+    See LongLivedServer/LongLivedClient for more details.
+    """
+
+    long_lived_conns = IntParam(mandatory=True)
+    long_lived_conns_port = IntParam(default=20000)
+    long_lived_conns_per_ip = IntParam(default=20000)
+
+    def _generate_ip_endpoints(self, config):
```

Comment:
please reorder the methods top -> down in order of abstraction/importance as we have in other recipes/mixins...

Reply:
done